pkg/util/user/range.go

```diff
@@ -0,0 +1,148 @@
+package user
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Range errors
+var (
+	ErrParseRange   = errors.New("error parsing range; a range must be of one of the following formats: [n], [-n], [n-], [n:m] where n and m are positive numbers and m is greater than or equal to n")
+	ErrInvalidRange = errors.New("invalid range; a range must consist of positive integers and the upper bound must be greater than or equal to the lower bound")
+)
+
+// Range represents a range of user ids. It can be unbound at either end with a nil value
+// I both From and To are present, To must be greater than or equal to From. Bounds are inclusive
+type Range struct {
+	from *int
+	to   *int
+}
+
+// NewRange creates a new range with lower and upper bound
+func NewRange(from int, to int) (*Range, error) {
+	if to < from || from < 0 || to < 0 {
+		return nil, ErrInvalidRange
+	}
+	return &Range{
+		from: &from,
+		to:   &to,
+	}, nil
+}
+
+// NewRangeTo creates a new range with only the upper bound
+func NewRangeTo(to int) (*Range, error) {
+	if to < 0 {
+		return nil, ErrInvalidRange
+	}
+	return &Range{
+		to: &to,
+	}, nil
+}
+
+// NewRangeFrom creates a new range with only the lower bound
+func NewRangeFrom(from int) (*Range, error) {
+	if from < 0 {
+		return nil, ErrInvalidRange
+	}
+	return &Range{
+		from: &from,
+	}, nil
+}
+
+// ParseRange creates a Range from a given string
+func ParseRange(value string) (*Range, error) {
+	value = strings.TrimSpace(value)
+	if value == "" {
+		return &Range{}, nil
+	}
+	parts := strings.Split(value, "-")
+	switch len(parts) {
+	case 1:
+		num, err := strconv.Atoi(parts[0])
+		if err != nil {
+			return nil, ErrParseRange
+		}
+		return NewRange(num, num)
+	case 2:
+		if parts[0] == "" {
+			to, err := strconv.Atoi(parts[1])
```

Comment:
I'm thinking, you're doing the int parsing over and over before each `NewRange*` method, how about move that logic there?